### ISSUE TITLE
Fatal error in AbstractLocalVariable

### ISSUE THREAD

**Issue by a-menshchikov** [phpmd/phpmd#816]
- PHPMD version: 2.9.0
- PHP Version: 7.4.9
- Installation type: composer
- Operating System / Distribution & Version: Debian 10.5

### Current Behavior
```
PHP Notice:  Undefined offset: 0 in /usr/src/app/vendor/phpmd/phpmd/src/main/php/PHPMD/Rule/AbstractLocalVariable.php on line 271
PHP Fatal error:  Uncaught Error: Call to a member function isPassedByReference() on null in /usr/src/app/vendor/phpmd/phpmd/src/main/php/PHPMD/Rule/AbstractLocalVariable.php:271
```

My phpmd-ruleset.xml:
```
<?xml version="1.0"?>
<ruleset name="Softline PHPMD rule set"
         xmlns="http://pmd.sf.net/ruleset/1.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sf.net/ruleset/1.0.0 http://pmd.sf.net/ruleset_xml_schema.xsd"
         xsi:noNamespaceSchemaLocation="http://pmd.sf.net/ruleset_xml_schema.xsd">

    <description>Softline Portal PHPMD rule set</description>

    <rule ref="rulesets/cleancode.xml/ElseExpression" />
    <rule ref="rulesets/cleancode.xml/DuplicatedArrayKey" />
    <rule ref="rulesets/cleancode.xml/UndefinedVariable" />

    <rule ref="rulesets/codesize.xml/CyclomaticComplexity">
        <properties>
            <property name="reportLevel" value="7" />
        </properties>
    </rule>
    <rule ref="rulesets/codesize.xml/NPathComplexity">
        <properties>
            <property name="minimum" value="40" />
        </properties>
    </rule>

    <rule ref="rulesets/codesize.xml/ExcessiveClassComplexity" />

    <rule ref="rulesets/controversial.xml/CamelCaseParameterName">
        <properties>
            <property name="allow-underscore" value="true" />
        </properties>
    </rule>
    <rule ref="rulesets/controversial.xml/CamelCaseVariableName">
        <properties>
            <property name="allow-underscore" value="true" />
        </properties>
    </rule>

    <rule ref="rulesets/design.xml">
        <exclude name="CouplingBetweenObjects" />
    </rule>

    <rule ref="rulesets/naming.xml">
        <exclude name="ShortVariable" />
        <exclude name="LongVariable" />
        <exclude name="ShortMethodName" />
    </rule>

</ruleset>
```

**Comment by kylekatarnls** [comment]
Thanks for the report. I did not get this on my projects. I will try to guess what syntax can trigger this.

**Comment by chelsEg** [comment]
I have the same error!

`Fatal error: Uncaught Error: Call to a member function isPassedByReference() on null in /path-to-vendor/phpmd/phpmd/src/main/php/PHPMD/Rule/AbstractLocalVariable.php:271
Stack trace:
`

<?xml version="1.0"?>
<ruleset name="PMD ruleset" xmlns="http://pmd.sf.net/ruleset/1.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sf.net/ruleset/1.0.0 http://pmd.sf.net/ruleset_xml_schema.xsd"
         xsi:noNamespaceSchemaLocation=" http://pmd.sf.net/ruleset_xml_schema.xsd">
    <description>PMD ruleset</description>

    <rule ref="rulesets/cleancode.xml">
        <exclude name="StaticAccess"/>
    </rule>
    <rule ref="rulesets/codesize.xml">
        <exclude name="TooManyMethods"/>
        <exclude name="TooManyPublicMethods"/>
    </rule>
    <rule ref="rulesets/codesize.xml/TooManyMethods">
        <properties>
            <property name="maxmethods" value="25" description="The method count reporting threshold."/>
        </properties>
    </rule>
    <rule ref="rulesets/codesize.xml/TooManyPublicMethods">
        <properties>
            <property name="maxmethods" value="15" description="The method count reporting threshold."/>
        </properties>
    </rule>
    <rule ref="rulesets/controversial.xml"/>
    <rule ref="rulesets/design.xml">
        <exclude name="NumberOfChildren"/>
    </rule>
    <rule ref="rulesets/design.xml/NumberOfChildren">
        <properties>
            <property name="minimum" value="25" description="Maximum number of acceptable child classes."/>
        </properties>
    </rule>
    <rule ref="rulesets/naming.xml">
        <exclude name="LongVariable"/>
        <exclude name="ShortVariable"/>
        <exclude name="BooleanGetMethodName"/>
    </rule>
    <rule ref="rulesets/unusedcode.xml">
    </rule>
</ruleset>

**Comment by thefrosty** [comment]
We have the same error. PHP 7.3.21.

**Comment by kylekatarnls** [comment]
It will be fixed in 2.9.1 thanks to @a-menshchikov. You can yet try the fix requiring `dev-master`.

**Comment by char101** [comment]
I already explained the problem a month ago https://github.com/phpmd/phpmd/issues/813 . The extra arguments in a function that takes a variable number of arguments are not reflectable.

**Comment by kylekatarnls** [comment]
Thank you @char101 I'm sorry for missing your issue. As you may see, we have still a lot of issues to catch up and it's easier for us to apply fixes when they are submitted as a pull-request.

**Comment by AdamVyborny** [comment]
> It will be fixed in 2.9.1 thanks to @a-menshchikov. You can yet try the fix requiring `dev-master`.

When it will be released?
